src/renderers/dom/shared/dangerousStyleValue.js

```diff
@@ -43,45 +43,12 @@ function dangerousStyleValue(name, value, component) {
     return '';
   }
 
-  var isNonNumeric = isNaN(value);
-  if (isNonNumeric || value === 0 || value === '0' ||
-      isUnitlessNumber.hasOwnProperty(name) && isUnitlessNumber[name]) {
-    return '' + value; // cast to string
+  if (typeof value === 'number' && value !== 0 &&
+      !(isUnitlessNumber.hasOwnProperty(name) && isUnitlessNumber[name])) {
+    return value+'px'; // Hack appends 'px' to unitless numbers
```

Comment:
Someone (Jordan?) called it a hack.  The fact that the code presumes `px` instead of `%` or any other unit (`em`, `ex`, `cm`, `mm`, `in`, `pt`, `pc`, etc).